bin/tx-prover/src/proxy/mod.rs

```diff
@@ -154,77 +179,70 @@ impl ProxyHttp for LoadBalancer {
-    /// and return the [HttpPeer]. The peer is configured with timeouts, and HTTP/2.
+    /// Here we enqueue the request and wait for it to be at the front of the queue and a worker
+    /// becomes available. We then set the SNI, timeouts, and enable HTTP/2.
     ///
     /// Note that the request is not removed from the queue here. It will be returned later in
     /// [Self::logging()] once the worker processes the it.
     async fn upstream_peer(
         &self,
-        session: &mut Session,
+        _session: &mut Session,
         ctx: &mut Self::CTX,
     ) -> Result<Box<HttpPeer>> {
-        if ctx.tries > 0 {
-            Self::remove_request_from_queue(Self::get_request_id(session)?).await;
-        }
+        let request_id = ctx.request_id;
 
-        // Select the worker in a round-robin fashion
-        let worker = self.lb.select(b"", 256).ok_or(Error::new_str("Worker not found"))?;
-
-        // Read request ID from headers
-        let request_id = {
-            let id = Self::get_request_id(session)?;
-            id.to_string()
-        };
-
-        // Enqueue the request ID in the worker queue
-        // We use a new scope to release the lock after the operation
-        {
-            let mut ctx_guard = QUEUES.write().await;
-            let worker_queue = ctx_guard.entry(worker.clone()).or_insert_with(Vec::new);
-
-            // Limit queue length
-            if worker_queue.len() >= self.max_queue_items {
-                Self::create_queue_full_response(session).await?;
-            }
-
-            worker_queue.push(request_id.to_string());
-        }
+        // Add the request to the queue.
+        QUEUE.enqueue(request_id).await;
 
         // Wait for the request to be at the front of the queue
         loop {
-            // We use a new scope for each iteration to release the lock
+            // We use a new scope for each iteration to release the lock before sleeping
             {
-                let ctx_guard = QUEUES.read().await;
-                if let Some(worker_queue) = ctx_guard.get(&worker) {
-                    if worker_queue[0] == request_id {
-                        break;
-                    }
-                } else {
-                    return Err(Error::new_str("Worker not found"));
+                // The request is at the front of the queue.
+                if QUEUE.peek().await.expect("Queue should not be empty") != request_id {
+                    continue;
+                }
+
+                // Check if there is an available worker
+                if let Some(worker) = self.get_available_worker().await {
+                    ctx.set_worker(worker);
+                    info!("Worker picked up the request with ID: {}", request_id);
+                    break;
                 }
+                info!("All workers are busy");
             }
             tokio::time::sleep(Duration::from_millis(100)).await;
```

Comment:
I would maybe reduce the sleep duration to something like 20ms. Also, may be good to create a constant for this value.

Reply:
You are right, not quite sure if I should use a constant or this should be defined in the configuration too. What do you think?

I will use a constant for now, would if we prefer the other approach I can do it easily.

Reply:
Yes, adding to config would be better - but let's do it in a separate PR.

Also not for this PR, but we should be able to update `pingroa` dependencies to 0.4 version.

Reply:
I open up this issue to track this: https://github.com/0xPolygonMiden/miden-base/issues/986